PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/samples/MavenProjectGenerator.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ * <p/>
+ * All rights reserved.
+ * <p/>
+ * MIT License
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.hdinsight.projects.samples;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.microsoft.azure.hdinsight.common.StreamUtil;
+import com.microsoft.azure.hdinsight.projects.HDInsightTemplatesType;
+import com.microsoft.azure.hdinsight.projects.SparkVersion;
+import com.intellij.openapi.util.io.FileUtil;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import org.jetbrains.idea.maven.project.MavenProjectsManager;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+
+public class MavenProjectGenerator {
+    private Module module;
+    private HDInsightTemplatesType templatesType;
+    private SparkVersion sparkVersion;
+
+    public MavenProjectGenerator(@NotNull Module module,
+                                 @NotNull HDInsightTemplatesType templatesType,
+                                 @NotNull SparkVersion sparkVersion) {
+        this.module = module;
+        this.templatesType = templatesType;
+        this.sparkVersion = sparkVersion;
+    }
+
+    public void generate() {
+        String root = ProjectSampleUtil.getRootOrSourceFolder(this.module, false);
+
+        try {
+            createDirectories(root);
+            createPom(root);
+            copySamples(root);
+            importMavenProject();
+        } catch (Exception e) {
+            DefaultLoader.getUIHelper().showError("Failed to create project", "Create Sample Project");
+            e.printStackTrace();

```

Comment:
remove this one since the DefaultLoader has been called to record the exception?